btcpp_ros2_samples/src/sample_bt_executor.cpp

```diff
@@ -44,11 +44,13 @@ class MyActionServer : public BT::TreeExecutionServer
     logger_cout_ = std::make_shared<BT::StdCoutLogger>(tree);
   }
 
-  void onTreeExecutionCompleted(BT::NodeStatus status, bool was_cancelled) override
+  std::optional<std::string> onTreeExecutionCompleted(BT::NodeStatus status,
```

Comment:
So are you thinking this string would be a way to return a json encoded string back to the caller that's scraped off the global blackboard?